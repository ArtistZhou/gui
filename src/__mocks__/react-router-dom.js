import React from 'react';
import { compose, setDisplayName } from 'recompose';
import * as rrd from 'react-router-dom';
// Just render plain div with its children
rrd.BrowserRouter = compose(setDisplayName('BrowserRouter'))(({ children }) => <div>{children}</div>);

module.exports = rrd;
