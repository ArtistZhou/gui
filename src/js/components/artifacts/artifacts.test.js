import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import configureStore from 'redux-mock-store';
import Artifacts from './artifacts';

const mockStore = configureStore([thunk]);

describe('Artifacts Component', () => {
  let store;
  beforeEach(() => {
    store = mockStore({
      releases: {
        byId: {},
        selectedArtifact: null,
        selectedRelease: null,
        showRemoveDialog: false,
        uploadProgress: 0
      },
      users: {
        onboarding: {
          complete: false
        }
      }
    });
  });

  it('renders correctly', () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <Provider store={store}>
            <Artifacts />
          </Provider>
        </MemoryRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
