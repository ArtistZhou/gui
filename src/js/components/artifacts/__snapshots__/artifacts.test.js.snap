// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Artifacts Component renders correctly 1`] = `
<div
  style={
    Object {
      "height": "100%",
    }
  }
>
  <div
    className="repository"
  >
    <div
      className="repository-list"
    >
      <div
        className="margin-bottom-small"
      >
        <h4>
          Releases
        </h4>
        <div
          className="search"
        >
          <input
            onChange={[Function]}
            placeholder="Filter by name"
            type="search"
            value=""
          />
        </div>
        <p
          className="margin-top muted align-center margin-right"
        >
          There are no Releases yet
        </p>
      </div>
      <div
        className="loaderContainer"
      >
        <div
          className="loader"
        >
          <span
            className="dot dot_1"
          />
          <span
            className="dot dot_2"
          />
          <span
            className="dot dot_3"
          />
          <span
            className="dot dot_4"
          />
        </div>
      </div>
    </div>
    <div
      className="flexbox centered"
      style={
        Object {
          "height": "50%",
          "width": "100%",
        }
      }
    >
      <div
        className="loaderContainer"
      >
        <div
          className="loader"
        >
          <span
            className="dot dot_1"
          />
          <span
            className="dot dot_2"
          />
          <span
            className="dot dot_3"
          />
          <span
            className="dot dot_4"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
